### PR TITLE
homework for meal sharing app react2 week2

### PR DESCRIPTION
**Pull request by aladdin91**
this is my h/w for meal sharing app react2 week2

### CODE REVIEW

File: src/client/components/ReservationForm.jsx [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+    } catch (err) {
+      setError(err.message);
+      alert(`Error: ${error}`);
+    }
+  };
+
+  return (
+    <>
+      <form className="form-card" onSubmit={handleSubmit}>
+        {/* i have to incloud id because the database does not increse the
```
Comment (shpomp) [inline src/client/components/ReservationForm.jsx:51]: 👀

File: src/client/components/MealList.jsx [hunk 2]
```diff
@@ -1,3 +1,22 @@
...
+  return (
+    <div>
+      <Navbar />
+      <h2>List of meals</h2>
+      <div className="container">
+        {meals.map((meal) => (
+          <Meal key={meal.id} meal={meal} />
+        ))}
+      </div>
+      <Footer />
```
Comment (shpomp) [inline src/client/components/MealList.jsx:19]: The whole content of this file is a bit of a mess. Please clean it up 🙏🏽

File: src/client/components/MealDetail.jsx [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+import Footer from "./Footer";
+import ShowReviews from "./ShowReviews";
+import ReservationForm from "./ReservationForm";
+import PostReviewForm from "./PostReviewForm";
+const mealsUrl = process.env.REACT_APP_MEALS_URL;
+
+function MealDetail() {
+  const { id } = useParams();
+  const { meals } = useContext(MealContext);
+  const meal = meals.find((meal) => meal.id === Number(id));
```
Comment (shpomp) [inline src/client/components/MealDetail.jsx:14]: Such a component that is meant to display details of a single meal - if I understood correctly - should definitely have a single meal passed into it as props.
Why would it need to know about all the meals and filter them? 🤔

File: src/client/components/Meal.jsx [hunk 4]
```diff
@@ -0,0 +1,30 @@
+import React from "react";
+import { Link } from "react-router-dom";
+import Button from "./Button";
+
+function Meal({ meal }) {
```
Comment (shpomp) [inline src/client/components/Meal.jsx:5]: Why do we have this and also the MealDetail?

File: src/backend/api/reviews.js [hunk 5]
```diff
@@ -17,7 +17,20 @@ router.get('/', async (req, res) => {
...
+    const meal_id = req.params.id;
+    const meal_reviews = await knex('review')
+      .select('*')
+      .where('meal_id', meal_id);
+    meal_reviews.length === 0 ?
+      res.json({ "message": " no reviews found for this meal ID" }) :
+      res.json(meal_reviews)
+  } catch (err) {
+    console.error(err);
+    res.status(500).send(err)
```
Comment (shpomp) [inline src/backend/api/reviews.js:31]: Are you sure that 500 is the right status here? 🤓

**Review by shpomp — CHANGES_REQUESTED** [review]
Overall quite a good job! 👏🏽 
There are some not-so-weel organised parts though that makes it difficult to read and understand the code and the app structure 🙏🏽

**Comment by aladdin91** [comment]
thanks a lot Karo for the review 🙏, i fixed it according to the review in the finel homework

**Pull request merged**